### ISSUE TITLE
V4.2.10:  Data Sharing Screen Very Slow

### ISSUE THREAD

**Issue by jones139** [OpenSeizureDetector/Android_Pebble_SD#199]
If the user has a lot of data in the online database, the data sharing screen is very slow and appears unresponsive - the Android system prompts whether to wait or close the app.

The solution is probably to limit the number of events returned by the web API to a manageable number?

I will probably not fix this in the Android Java code, because I am thinking of moving the event data editing to a web application rather than a native Android Java activity so that you can edit on a computer as well as on a phone.

**Comment by pmithrandir** [comment]
What about grouping events on the server before sending them to the app?When Arthur make a seizure of 5 minutes... I have around 50 events to fill, all being about the same real case in the end.If the API were grouping them and sending only one thing to enrich... It would be easier and lighter.We could work on what could make multiple event one. I know you have a gripping per 2 minutes, but maybe we can find a more accurate way.That part being in python, it's more my thing. I could work on it this summer if you want.Pierre Sent from Android deviceLe 23 juin 2024 20:48, Graham Jones ***@***.***> a écrit :
If the user has a lot of data in the online database, the data sharing screen is very slow and appears unresponsive - the Android system prompts whether to wait or close the app.
The solution is probably to limit the number of events returned by the web API to a manageable number?
I will probably not fix this in the Android Java code, because I am thinking of moving the event data editing to a web application rather than a native Android Java activity so that you can edit on a computer as well as on a phone.

**Comment by jones139** [comment]
Yes, when I implement a fix, I'll include grouping events so you do not need to edit each one separately - I'm not sure if the grouping will happen on the server or the front end at the moment though - it might be best to do it on the front end so the user has the option of un-grouping in case two different events happened in close succession.

**Comment by jones139** [comment]
I have discovered that there are two issues here.   One is likely to be the amount of data returned from the web API, but the other is that there is a delay between opening the  data sharing screen and being able to do anything.   It looks like this might be the query to display how many datapoints there are in the local database.   As I am probably the only one who would be interested in this, I'll remove that field from the screen and see if that improves matters.

I'll still do grouping though, honest...it is just more difficult to do!

**Comment by pmithrandir** [comment]
But that's a great thing... Because it will also improve the user XP and normally the amount of people sharing willingly value to enrich the data.Thank you very much for that.Sent from Android deviceLe 9 févr. 2025 18:52, Graham Jones ***@***.***> a écrit :
I have discovered that there are two issues here.   One is likely to be the amount of data returned from the web API, but the other is that there is a delay between opening the  data sharing screen and being able to do anything.   It looks like this might be the query to display how many datapoints there are in the local database.   As I am probably the only one who would be interested in this, I'll remove that field from the screen and see if that improves matters.
I'll still do grouping though, honest...it is just more difficult to do!

**Comment by jones139** [comment]
Fixed in V4.2.11
